changelog/fix_frozen_literal_detection_in_lint_void.md

```diff
@@ -0,0 +1 @@
+* [#13134](https://github.com/rubocop/rubocop/pull/13134): Fix false negative for `Lint/Void` when using using frozen literals. ([@vlad-pisanov][])
```

Comment:
I don't think this is entirely correct. These are not literals anymore, I think better would be something like "when calling `freeze` on literals`.

The offense message for these is also a bit meh but I wouldn't bother changing that.